thorntail-runner/src/main/java/org/wildfly/swarm/runner/WarBuilder.java

```diff
@@ -54,9 +55,40 @@ private WarBuilder(ZipOutputStream output, List<String> classesDirs, List<File>
     }
 
     private void build() {
-        classesDirs.forEach(this::addClassesToWar);
+        List<String> warDirs = classesDirs.stream()
+                .filter(this::isWar)
+                .collect(Collectors.toList());
+
+        classesDirs.stream()
+                .filter(d -> !isWar(d))
+                .map(this::buildJar)
+                .forEach(jars::add);
+
+        warDirs.forEach(this::addClassesToWar);
         addWebAppResourcesToWar();
-        jars.stream().forEach(this::addJarToWar);
+
+        jars.forEach(this::addJarToWar);
+    }
+
+    private File buildJar(String classesDir) {
+        try {
+            File jar = File.createTempFile("thorntail-user-jar", ".jar");
+
+            try (FileOutputStream fos = new FileOutputStream(jar);
+                 ZipOutputStream out = new ZipOutputStream(fos)) {
+                addClassesToXar(out, "", new File(classesDir));
+            }
+            return jar;
```

Comment:
Where are these files created? I'm thinking if we should somehow take care of deleting them or not.

Reply:
Good catch, I forgot about `deleteOnExit` for it.
They are created in `/tmp`